libs/build_dod_dataset.py

```diff
@@ -203,33 +327,43 @@ def get_usa_by_states_df():
 def join_and_output_shapefile(df, shp_reader, pivot_shp_field, pivot_df_column, shp_writer):
     blacklisted_fields = ['OBJECTID', 'Province/State', 'Country/Region', 'Last Update',
         'Latitude', 'Longitude', 'County', 'State/County FIPS Code', 'Combined Key']
+    non_integer_fields = ['Intervention', 'PEAK-HOSP', 'PEAK-DEATHS']
 
     fields = [field for field in df.columns if field not in blacklisted_fields]
 
     shp_writer.fields = shp_reader.fields # Preserve fields that come from the census
 
     for field_name in fields:
-        if field_name == 'Intervention': # Intervention is currently our only non-integer field
+        if field_name in non_integer_fields: # Intervention is currently our only non-integer field
             shp_writer.field(field_name, 'C', size=32)
         else:
             shp_writer.field(field_name, 'N', size=14)
 
     shapeRecords = shp_reader.shapeRecords()
-    # if you are using a local copy of the data, LFS truncates the records  
+    # if you are using a local copy of the data, LFS truncates the records
     assert len(shapeRecords) >= 50
 
+    # Just adding some understanding of errors
+    failed_dictionary = {}
+
     for shapeRecord in shapeRecords:
         try:
+            # Gets the row of the dataframe that matches the FIPS codes for a state/county
             row = df[df[pivot_df_column] == shapeRecord.record[pivot_shp_field]].iloc[0]
-        except:
+        except Exception as e:
+            state_fips = shapeRecord.record[pivot_shp_field][:2] # state fips is the first two chars of the state/county fips
```

Comment:
the FIPS for the state is also available as a separate entry (if you dont want to string split)